metasynth/var.py

```diff
@@ -92,28 +104,32 @@ def detect(cls, series_or_dataframe: Union[pd.Series, pd.DataFrame],
         MetaVar:
             It returns a meta data variable of the correct type.
         """
-        if isinstance(series_or_dataframe, pd.DataFrame):
-            return [MetaVar.detect(series_or_dataframe[col])
-                    for col in series_or_dataframe]
+        if isinstance(series_or_dataframe, (pl.DataFrame, pd.DataFrame)):
+            if isinstance(series_or_dataframe, pd.DataFrame):
+                return [MetaVar.detect(series_or_dataframe[col])
+                        for col in series_or_dataframe]
+            return [MetaVar.detect(series) for series in series_or_dataframe]
 
-        series = series_or_dataframe
-        return cls(cls.get_var_type(pd.api.types.infer_dtype(series)), series,
+        series = _to_polars(series_or_dataframe)
+
+        try:
+            var_type = pl.datatypes.dtype_to_py_type(series.dtype).__name__
+        except NotImplementedError:
+            var_type = pl.datatypes.dtype_to_ffiname(series.dtype)
+        return cls(cls.get_var_type(var_type), series,
                    description=description, prop_missing=prop_missing)
 
     @staticmethod
     def get_var_type(pandas_dtype: str) -> str:
         """Convert pandas dtype to MetaSynth variable type."""
         convert_dict = {
-            "categorical": "categorical",
-            "string": "string",
-            "integer": "discrete",
-            "floating": "continuous",
-            "mixed-integer-float": "continuous",
-            "empty": "continuous",
+            "int": "discrete",
+            "float": "continuous",
             "date": "date",
-            "datetime64": "datetime",
             "datetime": "datetime",
             "time": "time",
+            "str": "string",
+            "categorical": "categorical"
         }
         try:
```

Comment:
I would expect that we don't use this anymore? this should be done by pandas -> polars conversion function and then we only have to maintain one vartype conversion list (namely polars to metasynth).

Reply:
This is the polars -> metasynth conversion. :) I updated the names/code.